src/main/java/uk/gov/hmcts/reform/sscs/migration/service/CaseManagementLocationService.java

```diff
@@ -0,0 +1,39 @@
+package uk.gov.hmcts.reform.sscs.migration.service;
+
+import static java.util.Objects.nonNull;
+import static org.apache.commons.lang3.StringUtils.isNotBlank;
+
+import java.util.Optional;
+import lombok.RequiredArgsConstructor;
+import org.springframework.stereotype.Service;
+import uk.gov.hmcts.reform.sscs.ccd.domain.CaseManagementLocation;
+import uk.gov.hmcts.reform.sscs.model.CourtVenue;
+import uk.gov.hmcts.reform.sscs.service.RefDataService;
+
+@Service
+@RequiredArgsConstructor
+public class CaseManagementLocationService {
+
+    private final RefDataService refDataService;
+    private final RpcVenueService rpcVenueService;
+
+    public Optional<CaseManagementLocation> retrieveCaseManagementLocation(String processingVenue, String postcode) {
+        if (isNotBlank(processingVenue)
+            && isNotBlank(postcode)) {
+
+            CourtVenue courtVenue = refDataService.getVenueRefData(processingVenue);
+            String rpcEpimsId = rpcVenueService.retrieveRpcEpimsIdForPostcode(postcode);
+
+            if (nonNull(courtVenue)
+                && isNotBlank(courtVenue.getRegionId())
+                && isNotBlank(rpcEpimsId)) {
+                return Optional.of(CaseManagementLocation.builder()
+                    .baseLocation(rpcEpimsId)
+                    .region(courtVenue.getRegionId())
+                    .build());
+            }
+        }
```

Comment:
Better to reverse this to get out early.  I.e. if blank, then return

Reply:
Disagree.. how exactly will this make a difference?